src/v/cloud_storage/partition_probe.h

```diff
@@ -20,12 +20,26 @@ namespace cloud_storage {
 
 class partition_probe {
 public:
-    using hist_t = log_hist_internal;
-
     explicit partition_probe(const model::ntp& ntp);
 
     void add_bytes_read(uint64_t read) { _bytes_read += read; }
     void add_records_read(uint64_t read) { _records_read += read; }
+    void chunk_size(uint64_t size) { _chunk_size = size; }
+
+private:
+    uint64_t _bytes_read = 0;
+    uint64_t _records_read = 0;
+    uint64_t _chunk_size = 0;
+
+    ssx::metrics::metric_groups _metrics
+      = ssx::metrics::metric_groups::make_internal();
+};
+
+class ts_read_path_probe {
```

Comment:
nit: maybe it would make sense to rename this file to something more generic now that it contains various probe types.

Reply:
done